packages/controller/lib/setup.js

```diff
@@ -2725,37 +2729,39 @@ async function delStates() {
     return keys ? keys.length : 0;
 }
 
-function cleanDatabase(isDeleteDb, callback) {
+/**
+ * Cleans the database
+ *
+ * @param {boolean} isDeleteDb - if whole db should be destroyed
+ * @return {Promise<number>}
+ */
+async function cleanDatabase(isDeleteDb) {
     if (isDeleteDb) {
-        objects.destroyDB(async () => {
-            // Clean up states
-            const keysCount = await delStates();
-
-            if (callback) {
-                return void callback(keysCount);
-            }
-        });
+        await objects.destroyDBAsync();
+        // Clean up states
+        return delStates();
```

Comment:
return promise here, but below return count only??

Reply:
This works (async function!), but isn't consistent, and the code below is more obvious IMO.

Reply:
yes basically the same as assigning to const and then returining, just an unnecessary step but yes just seeing it right now that its in two steps below and if you find it better readable lets to it like that